packages/fxa-settings/src/pages/Authorization/container.tsx

```diff
@@ -0,0 +1,161 @@
+  account: ReturnType<typeof currentAccount>,
+  authClient: ReturnType<typeof useAuthClient>
+) => {
+  const relierAccount: RelierAccount = {
+    uid: account?.uid!,
+    email: account?.email!,
+    sessionToken: account?.sessionToken!,
+    verifyIdToken: authClient.verifyIdToken,
+    isDefault: () => isDefault(account || {}),
+  };
+  return relierAccount;
+};
+
+/**
+ * Unlike the other containers, this one does not attempt to render anything.
+ * It handles a 'prompt=none' authorization or hand off to the signin routes.
+ */
+const AuthorizationContainer = ({
+  integration,
+}: {
+  integration: Integration;
+} & RouteComponentProps) => {
+  const [oauthError, setOauthError] = useState<AuthError | OAuthError | null>(
+    null
+  );
+  const navigate = useNavigate();
+  const authClient = useAuthClient();
+  const location = useLocation() as ReturnType<typeof useLocation>;
+  const session = useSession();
+  const { finishOAuthFlowHandler, oAuthDataError } = useFinishOAuthFlowHandler(
+    authClient,
+    integration
+  );
+  if (oAuthDataError) {
+    setOauthError(oAuthDataError);
+  }
+
+  const promptNoneHandler = useCallback(async () => {
+    debugger;
+    const account = currentAccount();
+    const relierAccount = convertToRelierAccount(account, authClient);
+
+    try {
+      await (integration as OAuthWebIntegration).validatePromptNoneRequest(
+        relierAccount
+      );
+
+      const { data, error } = await cachedSignIn(
+        account?.sessionToken!,
+        authClient,
+        cache,
+        session
+      );
+
+      if (error === AuthUiErrors.SESSION_EXPIRED) {
+        throw new OAuthError('PROMPT_NONE_NOT_SIGNED_IN');
+      }
+
+      if (error) {
+        throw error;
+      }
+
+      if (!data?.verified) {
+        throw new OAuthError('PROMPT_NONE_UNVERIFIED');
+      }
+
+      if (data) {
+        const navigationOptions = {
+          email: account?.email!,
+          signinData: {
+            verified: data.verified,
+            verificationMethod: data.verificationMethod,
+            verificationReason: data.verificationReason,
+            uid: data.uid,
+            sessionToken: account?.sessionToken!,
+          },
+          integration,
+          redirectTo: integration.data.redirectTo,
+          finishOAuthFlowHandler,
+          queryParams: location.search,
+        };
+
+        const { error: navError } = await handleNavigation(navigationOptions);
+
+        if (navError) {
+          throw navError;
+        }
+      }
+    } catch (err) {
+      setOauthError(err);
+    }
+  }, [
+    authClient,
+    finishOAuthFlowHandler,
+    integration,
+    location.search,
+    session,
+  ]);
+
+  useEffect(() => {
+    if (
+      integration instanceof OAuthWebIntegration &&
+      integration.wantsPromptNone()
+    ) {
+      promptNoneHandler();
+      return;
+    }
+
+    const urlSearchParams = new URLSearchParams(location.search);
+    urlSearchParams.delete('showReactApp');
```

Comment:
Once `/oauth` is handled in React, this and the `hardNavigate` calls below can be updated to navigate to `/oauth` in React.